aspnetcore/blazor/tutorials/movie-database-app/part-8.md

```diff
@@ -0,0 +1,292 @@
+
+```diff
+- <PropertyColumn Property="movie => movie.Title" />
++ <PropertyColumn Property="movie => movie.Title" Sortable="true" />
+```
+
+Run the app and navigate to the movies `Index` page.
+
+You can sort the `QuickGrid` by movie title by selecting the **Title** column.
+
+The component is *interactive*. The page doesn't reload for sorting to occur. The sorting is performed live over the SignalR connection between the browser and the server, where the sorting operation is performed on the server with the rendered result sent back to the client for the browser to display.
+
+## Use C# code and interactivity to search by title
+
+In an earlier part of the tutorial series, the `Index` component was modified to allow the user to filter movies by title. This was accomplished by:
+
+* Adding an HTML form that issues a GET request to the server with the user's title search string as a query string field-value pair (for example, `?titleFilter=road+warrior` if the user searches for "`road warrior`"):
+
+  ```html
+  <form action="/movies">
+      <input type="text" name="titleFilter" />
+      <input type="submit" value="Search" />
+  </form>
+  ```
+
+* Adding code to the component that obtains the title search string from the query string and uses it to filter the database's records:
+
+  ```csharp
+  [SupplyParameterFromQuery]
+  public string? TitleFilter { get; set; }
+
+  protected override void OnParametersSet()
+  {
+      if (!string.IsNullOrEmpty(TitleFilter))
+      {
+          movies = DB.Movie.Where(
+              s => !string.IsNullOrEmpty(s.Title) ? 
+                  s.Title.Contains(TitleFilter) : false);
+      }
+      else
+      {
+          movies = DB.Movie;
+      }
+  }
+  ```
+
+The preceding approach is effective for a component that adopts static SSR, where the only interaction between the client and server is via HTTP requests. There was no live SignalR connection between the client and the server, and there was no way for the app on the server to process C# code *interactively* based on the user's actions in the component's UI and return content without a full-page reload.
+
+Now that the component is interactive, it can provide an improved user experience with Blazor features for binding and event handling, where full-page reloads aren't required to run C# on the server that interacts with the page's elements.
+
+Convert the `TitleFilter` property into a C# field because an interactive component doesn't require a filter string supplied by a user to reach the server via a query string and a GET request. Blazor can bind an HTML element's value to a C# field or property transparently over the underlying SignalR connection. Change the following code for the filter string, including the casing of the variable to match the convention for fields, which is camel case (`TitleFilter` to `titleFilter`):
+
+```diff
+- [SupplyParameterFromQuery]
```

Comment:
The downside of removing this query string parameter is that the search results are no longer addressable. Can we preserve that functionality while enabling interactivity?

Reply:
I have a message out to Mackinnon now asking about updating the URL when the search box is used interactively because it would be nice if after an interactive update the user can immediately bookmark the search (i.e., via the correct query string with URL-encoded search filter in the address bar).

Reply:
As discussed over email, it looks like we should remove the attribute as part of enabling interactivity.